packages/drivers/odsp-driver/src/getFileLink.ts

```diff
@@ -111,11 +115,14 @@ async function getFileLinkCore(
                 };
                 const response = await fetchHelper(url, requestInit);
                 additionalProps = response.commonSpoHeaders;
-                if (response.content.ok) {
-                    const sharingInfo = await response.content.json();
-                    return sharingInfo?.d?.directUrl as string;
+
+                const sharingInfo = await response.content.json();
+                const directUrl = sharingInfo?.d?.directUrl;
+                if (typeof directUrl !== "string") {
+                    // This will retry once in getWithRetryForTokenRefresh
+                    throwOdspNetworkError("malformedGetSharingInformationResponse", fetchIncorrectResponse);
```

Comment:
Just opened a bug for general usage of this pattern -  https://github.com/microsoft/FluidFramework/issues/8258
Ok to leave it to Mark to deal with it later :)